File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/cognito/src/main/java/com/example/cognito/DeleteUserPool.java
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/
package com.example.cognito;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.cognitoidentityprovider.CognitoIdentityProviderClient;
import software.amazon.awssdk.services.cognitoidentityprovider.model.CognitoIdentityProviderException;
import software.amazon.awssdk.services.cognitoidentityprovider.model.DeleteUserPoolRequest;
import software.amazon.awssdk.services.cognitoidentityprovider.model.DeleteUserPoolResponse;

/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */

public class DeleteUserPool {

    public static void main(String[] args) {
        final String usage = "\n" +
                "Usage:\n" +
                "    <userPoolId> \n\n" +
                "Where:\n" +
                "    userPoolId - the Id value given to your user pool.\n\n" ;

        if (args.length != 1) {
            System.out.println(usage);
            System.exit(1);
        }

        String userPoolId = args[0];
        CognitoIdentityProviderClient cognitoClient = CognitoIdentityProviderClient.builder()
                .region(Region.US_EAST_1)
                .build();

        deletePool(cognitoClient, userPoolId);
        cognitoClient.close();
    }

    public static void deletePool(CognitoIdentityProviderClient cognitoClient, String userPoolId ) {

        try {
            DeleteUserPoolRequest request = DeleteUserPoolRequest.builder()
                    .userPoolId(userPoolId)
                    .build();

            DeleteUserPoolResponse response = cognitoClient.deleteUserPool(request);
            System.out.println("User pool " + response.toString() + " deleted. ID: " + request.userPoolId());

        } catch (CognitoIdentityProviderException e){
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
     }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/cognito/src/main/kotlin/com/kotlin/cognito/DeleteUserPool.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.cognito

import aws.sdk.kotlin.services.cognitoidentityprovider.CognitoIdentityProviderClient
import aws.sdk.kotlin.services.cognitoidentityprovider.model.DeleteUserPoolRequest
import kotlin.system.exitProcess

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args: Array<String>){

    val usage = """
        Usage: <userPoolId>
    
        Where:
            userPoolId - the Id value of your user pool.
        """

      if (args.size != 1) {
          println(usage)
          exitProcess(0)
     }

    val userPoolId= args[0]
    delPool(userPoolId)
    }

suspend fun delPool(userPoolId:String) {

    val request = DeleteUserPoolRequest{
        this.userPoolId = userPoolId
    }

   CognitoIdentityProviderClient { region = "us-east-1" }.use { cognitoClient ->
            cognitoClient.deleteUserPool(request)
            print("$userPoolId was successfully deleted")
    }
 }
